From cd99561093e806e86e4086abe2bdc55eab2f91a9 Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Fri, 31 May 2024 11:04:14 +0200
Subject: [PATCH 0917/1695] drivers: stm32_i2c: inform when the non-secure
 compatible is used

Add i2c_secure in i2c_handle_s that is set to true when a secure
compatible is used. This is used to discriminate the
"st,stm32mp15-i2c-non-secure" compatible for PMIC usage when I2C is
non-secure.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git e6c68ddef453759242fa71de58f1ad746a504552 drivers: stm32_i2c: inform when the non-secure compatible is used]
Change-Id: I80de9bf459e88fb177f0e5dd87d8ff061dc2a974
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/384283
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
---
 core/drivers/stm32_i2c.c         | 4 +++-
 core/include/drivers/stm32_i2c.h | 2 ++
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/core/drivers/stm32_i2c.c b/core/drivers/stm32_i2c.c
index ea9975a1d..8a42fab3c 100644
--- a/core/drivers/stm32_i2c.c
+++ b/core/drivers/stm32_i2c.c
@@ -1674,8 +1674,10 @@ static TEE_Result stm32_i2c_probe(const void *fdt, int node,
 	init_data.analog_filter = true;
 	init_data.digital_filter_coef = 0;
 
-	if (compat_data == &secure_i2c)
+	if (compat_data == &secure_i2c) {
 		stm32_pinctrl_set_secure_cfg(pinctrl_active, true);
+		i2c_handle_p->i2c_secure = true;
+	}
 
 	if (pinctrl_apply_state(i2c_handle_p->pinctrl))
 		panic();
diff --git a/core/include/drivers/stm32_i2c.h b/core/include/drivers/stm32_i2c.h
index 42b26b0eb..726674b94 100644
--- a/core/include/drivers/stm32_i2c.h
+++ b/core/include/drivers/stm32_i2c.h
@@ -116,6 +116,7 @@ struct i2c_cfg {
  * @pinctrl: Pin control configuration for the I2C bus in active state
  * @pinctrl_sleep: Pin control configuration for the I2C bus in standby state
  * @mu: Protection on concurrent access to the I2C bus considering PM context
+ * @i2c_secure: Indicates that the I2C is secure
  */
 struct i2c_handle_s {
 	struct io_pa_va base;
@@ -130,6 +131,7 @@ struct i2c_handle_s {
 	struct pinctrl_state *pinctrl;
 	struct pinctrl_state *pinctrl_sleep;
 	struct mutex_pm_aware mu;
+	bool i2c_secure;
 };
 
 /*
-- 
2.47.2

